### PR TITLE
Update reth, revm, alloy

### PR DESCRIPTION
**Pull request by kpp**
# Description

Update: reth, revm, alloy
Remove: celestia

## Linked Issues
- Related to https://github.com/chainwayxyz/citrea/issues/637

## Testing
Describe how these changes were tested. If you've added new features, have you added unit tests?

## Docs
Describe where this code is documented. If it changes a documented interface, have the docs been updated?

### CODE REVIEW

File: crates/sovereign-sdk/adapters/risc0-bonsai/src/host.rs [hunk 1]
```diff
@@ -460,16 +460,19 @@ impl<'a> ZkvmHost for Risc0BonsaiHost<'a> {
 
                         // now we convert the snark_receipt to a full receipt
 
+                        use risc0_zkvm::sha::Digestible;
+                        let inner = InnerReceipt::Groth16(Groth16Receipt::new(
+                            snark_receipt.snark.to_vec(),
+                            receipt.claim().expect("stark_2_snark error, receipt claim"),
+                            risc0_zkvm::Groth16ReceiptVerifierParameters::default().digest(), // FIXME
```
Comment (kpp) [inline crates/sovereign-sdk/adapters/risc0-bonsai/src/host.rs:467]: Fix

**Review by kpp — COMMENTED** [review]
Notes for myself

File: crates/sovereign-sdk/full-node/sov-stf-runner/src/runner.rs [hunk 2]
```diff
@@ -7,7 +7,7 @@ use backoff::future::retry as retry_backoff;
 use backoff::ExponentialBackoffBuilder;
 use borsh::de::BorshDeserialize;
 use borsh::BorshSerialize as _;
-use jsonrpsee::core::Error;
+use jsonrpsee::core::client::Error as JsonRError;
```
Comment (eyusufatik) [inline crates/sovereign-sdk/full-node/sov-stf-runner/src/runner.rs:10]: lets call this JsonrpseeError maybe?

File: crates/sovereign-sdk/fuzz/fuzz_targets/share_deserialize.rs
Comment (eyusufatik) [inline crates/sovereign-sdk/fuzz/fuzz_targets/share_deserialize.rs:1]: why?


I mean we don't really use these much just want to know

File: Cargo.toml [hunk 3]
```diff
@@ -142,29 +147,33 @@ ethers-providers = { version = "2.0", default-features = false }
...
-alloy-sol-types = "0.6.4"
+alloy-rlp = { version = "0.3", default-features = false }
+alloy-primitives = { version = "0.7.4", default-features = false }
+alloy-sol-types = { version = "0.7.4", default-features = false }
 
-secp256k1 = { version = "0.27.0", default-features = false, features = [
+ed25519-dalek = { version = "2", default-features = false, features = [
+    "serde",
+] }
+secp256k1 = { version = "0.28.2", default-features = false, features = [
```
Comment (eyusufatik) [inline Cargo.toml:176]: is the patch defined below for secp256k1 still used?
given that this is 28.2 and thats 28?

File: crates/ethereum-rpc/src/lib.rs [hunk 4]
```diff
@@ -220,12 +219,12 @@ fn register_rpc_methods<C: sov_modules_api::Context, Da: DaService>(
         info!("eth module: eth_feeHistory");
         let mut params = params.sequence();
 
-        let block_count: U64HexOrNumber = params.next()?;
+        let block_count: Index = params.next()?;
```
Comment (eyusufatik) [inline crates/ethereum-rpc/src/lib.rs:222]: Was U64HexOrNumber replaced with Index type in reth?

this should parse both 30 and 0x1e

**Review by eyusufatik — CHANGES_REQUESTED** [review]

File: crates/sovereign-sdk/fuzz/fuzz_targets/share_deserialize.rs
Comment (kpp) [reply crates/sovereign-sdk/fuzz/fuzz_targets/share_deserialize.rs:1]: Because it uses celestia code.

Comment (eyusufatik) [reply crates/sovereign-sdk/fuzz/fuzz_targets/share_deserialize.rs:1]: appearently creates some dependency problems

File: Cargo.toml (hunk 3 above)
Comment (kpp) [reply Cargo.toml:176]: Yes

File: crates/ethereum-rpc/src/lib.rs (hunk 4 above)
Comment (kpp) [reply crates/ethereum-rpc/src/lib.rs:222]: https://reth.rs/docs/reth_rpc_types/struct.Index.html

Yes but it's usize now, not u64

File: crates/sovereign-sdk/full-node/sov-stf-runner/src/runner.rs (hunk 2 above)
Comment (kpp) [reply crates/sovereign-sdk/full-node/sov-stf-runner/src/runner.rs]: Done

File: crates/ethereum-rpc/src/lib.rs (hunk 4 above)
Comment (eyusufatik) [reply crates/ethereum-rpc/src/lib.rs:222]: checked it its fine

**Review by eyusufatik — APPROVED** [review]

**Pull request merged**